src/App.tsx

```diff
@@ -15,49 +16,124 @@ export const goodsFromServer = [
+          return goodA.length - goodB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const handleSortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const handleSortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const handleReversed = () => {
+    setReverse(!isReversed);
+  };
+
+  const handleReset = () => {
+    setReverse(false);
+    setSortType(SortType.NONE);
+  };
+
+  const sortAlphabetCN = cn('button is-info',
+    { 'is-light': sortType !== SortType.ALPHABET });
+
+  const sortBylengthCN = cn('button is-success',
+    { 'is-light': sortType !== SortType.LENGTH });
+
+  const reverseCN = cn('button is-warning', { 'is-light': !isReversed });
+
   return (
     <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+      <h1 className="title is-1">React List of Goods Hooks</h1>
+      <div className="wrapper">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sortAlphabetCN}
+            onClick={handleSortAlphabetically}
+          >
+            Sort alphabetically
+          </button>
 
-      <ul>
+          <button
+            type="button"
+            className={sortBylengthCN}
+            onClick={handleSortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={reverseCN}
+            onClick={handleReversed}
+          >
+            Reverse
+          </button>
+          {(sortType !== SortType.NONE || isReversed) && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={handleReset}
+            >
+              Reset
+            </button>
+          )}
+        </div>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {getReorderedGoods(goodsFromServer, { sortType, isReversed })
```

Comment:
better move getReorderedGoods(goodsFromServer, { sortType, isReversed } to variable